astropy/coordinates/sites.py

```diff
@@ -0,0 +1,203 @@
+
+# Observatory database and list of names:
+_site_db = None
+_site_names = []
+
+def _load_sites():
+    """
+    Load observatory database from data/observatories.json and parse them into
+    a dictionary in memory.
+    """
+    global _site_db, _site_names
+    _site_db = dict()
+    db = json.loads(get_pkg_data_contents('data/observatories.json'))
+    for site in db:
+        location = EarthLocation.from_geodetic(db[site]['longitude'],
+                                   db[site]['latitude'],
+                                   db[site]['elevation_meters'])
+        _site_names.append(db[site]['name'])
+        for alias in db[site]['aliases']:
+            _site_db[alias.lower()] = location
+
+def get_site(site_name):
+    """
+    Return an `~astropy.coordinates.EarthLocation` object for known observatory.
+
+    Use `~astropy.coordinates.get_site_names` to see a list of available
+    observatories.
+
+    Parameters
+    ----------
+    site_name : str
+        Name of the observatory.
+
+    Returns
+    -------
+    `~astropy.coordinates.EarthLocation`
+        The location of the observatory.
+
+    See Also
+    --------
+    get_site_names : the list of sites that this function can access
+    add_site : adds a new site to the list
+    remove_site : remove a site by name from the list of sites
+    """
+    if _site_db is None:
+        _load_sites()
+
+    if site_name.lower() not in _site_db.keys():
+        # If site name not found, find close matches and suggest them in error
+        close_names = get_close_matches(site_name, _site_db.keys())
+        close_names = sorted(close_names, key=lambda x: len(x))
+        if len(close_names) > 0:
+            errmsg = ("Site not in database. Use ``get_site_names()`` "
+                      "to see available sites. Did you mean: '{}'?".format(
+                      "', '".join(close_names)))
+        else:
+            errmsg = 'Site not in database.'
+        raise KeyError(errmsg)
+
+    return _site_db[site_name.lower()]
+
+def get_site_names(full_list=True):
+    """
+    Get list of names of observatories for use with
+    `~astropy.coordinates.get_site`.
+
+    Parameters
+    ----------
+    full_list : bool
+        Show full list observatory names and aliases (True), or just the list
+        of names (False)? Default to True.
+
+    Returns
+    -------
+    names : list of str
+        List of valid observatory names
+
+    See Also
+    --------
+    get_site : gets the `~astropy.coordinates.EarthLocation` for one of the
+               sites this returns.
+    get_site_names : the list of sites names that this function can access
+    remove_site : remove a site by name from the list of sites
+
+    """
+    if _site_db is None:
+        _load_sites()
+
+    if full_list:
+        return sorted(_site_db.keys())
+    else:
+        return sorted(_site_names)
+
+def add_site(site_names, location):
+    """
+    Add a site to the list of available observatories.
```

Comment:
Which makes me think, in addition to updating the DB from an online resource, this should include the ability to persist a custom DB to `~/.astropy` or elsewhere.  But something like that could be added later (though makes another good argument for reorganization into a class).